shaptools/hdb_connector/connectors/dbapi_connector.py

```diff
@@ -65,11 +65,16 @@ def query(self, sql_statement):
         self._logger.info('executing sql query: %s', sql_statement)
         try:
             with self._connection.cursor() as cursor:
-                cursor.execute(sql_statement)
-                result = base_connector.QueryResult.load_cursor(cursor)
+                # check for cursor.execute() == False
+                # fixes: "shaptools.hdb_connector.connectors.base_connector.QueryError: query failed: (0, 'No result set')"
+                if cursor.execute(sql_statement):
+                    result = base_connector.QueryResult.load_cursor(cursor)
+                else:
+                    result = base_connector.QueryResult([], ())
```

Comment:
This looks something that we should raise as an exception. If the query fails, we shouldn't just return an empty value. We should raise an exception and inform the user why this happened

Reply:
@arbulu89 I restructured the `try/except` blocks completely and there are now more verbose exceptions.